torch_geometric/nn/conv/wl_conv_continuous.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union
+
+from torch import Tensor
+from torch_sparse import SparseTensor, matmul
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.typing import Adj, OptPairTensor, OptTensor, Size
+from torch_geometric.utils import degree
+
+
+class WLConvContinuous(MessagePassing):
+    r"""The Weisfeiler Lehman operator from the `"Wasserstein
+    Weisfeiler-Lehman Graph Kernels"
+    <https://arxiv.org/abs/1906.01277>`_ paper.
+    Refinement is done though a degree-scaled mean aggregation and works on
+    nodes with continuous attributes:
+
+    .. math::
+        \mathbf{x}^{\prime}_i = \frac{1}{2}\big(\mathbf{x}_i +
+        \frac{1}{\textrm{deg}(i)}
+        \sum_{j \in \mathcal{N}(i)} e_{j,i} \cdot \mathbf{x}_j \big)
+
+    where :math:`e_{j,i}` denotes the edge weight from source node :obj:`j` to
+    target node :obj:`i` (default: :obj:`1`)
+
+    Args:
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+
+    Shapes:
+        - **input:**
+          node features :math:`(|\mathcal{V}|, F_{in})` or
+          :math:`((|\mathcal{V_s}|, F_{s}), (|\mathcal{V_t}|, F_{t}))`
+          if bipartite,
+          edge indices :math:`(2, |\mathcal{E}|)`,
+          edge weights :math:`(|\mathcal{E}|)` *(optional)*
+        - **output:** node features :math:`(|\mathcal{V}|, F_{out})` or
+          :math:`(|\mathcal{V}_t|, F_{out})` if bipartite
+    """
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        super().__init__(aggr='mean', **kwargs)
+
+    def forward(self, x: Union[Tensor, OptPairTensor], edge_index: Adj,
+                edge_weight: OptTensor = None, size: Size = None) -> Tensor:
+        """"""
+        if isinstance(x, Tensor):
+            x: OptPairTensor = (x, x)
+
+        # propagate_type: (x: OptPairTensor, edge_weight: OptTensor)
+        out = self.propagate(edge_index, x=x, edge_weight=edge_weight,
+                             size=size)
+        x_r = x[1]
+        deg = degree(edge_index[1]).view(-1, 1)
```

Comment:
I think this could be written as a custom `Aggregation`?

Reply:
Thanks for the feedback @Padarn. I'm not sure the `Aggregation` class works here since the convolution operation is defined recursively on the output from previous layers. Do you mean to incorporate a custom aggregation within the convolution class?

Reply:
My thought was to add a new `Aggregation` that could be used inside of message passing when aggregating messages, not as a single `Aggregation`. Does that make sense?

Reply:
@Padarn you mean a custom aggregation that takes care of weighting based on node degree? Isn‘t that what `mean` is already doing? Why do we actually need to perform normalization twice here?

Reply:
You are right, @rusty1s , the aggregation in this convolution operation could be done with the mean aggregator, I didn't know about it when I wrote this. Then it just becomes a regular GraphConv with a mean aggregation and an optional scaling on the neighbors.

Reply:
Right, `mean` does it too :-)

Reply:
I think we could do provide a `WeightedAggregation` with custom weights per element, and clean this further up. Happy to take this in a separate PR.